nesc/whip6/apps/HWTest/HWTestPrv.nc

```diff
@@ -0,0 +1,164 @@
+/**
+ * whip6: Warsaw High-performance IPv6.
+ *
+ * Copyright (c) 2017 Uniwersytet Warszawski
+ * All rights reserved.
+ */
+
+#include <driverlib/osc.h>
+#include "stdio.h"
+#include "string.h"
+#include "common.h"
+
+#define BUFFER_SIZE 50
+#define CBUFFER_SIZE 100
+#define TIMEOUT_MS 1024
+
+module HWTestPrv {
+    uses interface Boot;
+    uses interface BufferedRead;
+    uses interface BufferedWrite;
+    uses interface Led;
+    uses interface DimensionalRead<TDeciCelsius, int16_t> as ReadTemp;
+    uses interface Timer<TMilli, uint32_t>;
+
+    uses interface Init as LowInit;
+    uses interface RawFrame;
+    uses interface RawFrameSender as LowFrameSender;
+    uses interface RawFrameReceiver as LowFrameReceiver;
+}
+
+implementation {
+    uint16_t cmd_size = 1;
+    error_t result;
+    error_t init_status;
+    uint8_t_xdata buffer[BUFFER_SIZE];
```

Comment:
I am using this suffix because the APIs requires this datatype. Of course, internally this is just uint8_t. Why API call don't just require uint8_t then?